feature-libs/cart/base/components/clear-cart/clear-cart-dialog/clear-cart-dialog-component.service.ts

```diff
@@ -1,77 +1,82 @@
     protected launchDialogService: LaunchDialogService,
     protected globalMessageService: GlobalMessageService,
-    protected activeCartFacade: ActiveCartFacade
+    protected activeCartFacade: ActiveCartFacade,
+    protected multiCartFacade: MultiCartFacade,
+    protected userIdService: UserIdService,
+    protected eventService: EventService
   ) {}
 
+  /**
+   * Clear all entries and info of active cart
+   */
   clearActiveCart(): void {
-    this.isClearing$.next(true);
-    this.clearCart()
-      .pipe(take(1))
-      .subscribe(() => {
-        this.isClearing$.next(false);
-        this.onComplete();
+    this.activeCartFacade
+      .getActiveCartId()
+      .pipe(
+        withLatestFrom(this.userIdService.getUserId()),
+        take(1),
+        tap(([cartId, userId]) => {
+          this.multiCartFacade.deleteCart(cartId, userId);
+        }),
+        switchMap(() =>
+          merge(
+            this.eventService.get(DeleteCartSuccessEvent).pipe(mapTo(true)),
+            this.eventService.get(DeleteCartFailEvent).pipe(mapTo(false))
+          )
+        ),
+        tap(() => this.closeDialog('Close dialog after cart cleared')),
+        take(1)
+      )
+      .subscribe((success: boolean) => {
+        this.displayGlobalMessage(success);
       });
   }
 
-  clearCart(): Observable<boolean> {
-    return this.activeCartFacade.getEntries().pipe(
-      take(1),
-      tap((entries) => {
-        // Make copy and reverse entries[] to start at end of array
-        // since cart entries are shifted downwards with removeEntry()
-        entries
-          .slice()
-          .reverse()
-          .forEach((entry) => this.activeCartFacade.removeEntry(entry));
-      }),
-      switchMap(() => this.activeCartFacade.isStable()),
-      filter((data) => Boolean(data))
-    );
-  }
-
-  getClearingCartProgess(): Observable<boolean> {
-    return this.isClearing$.asObservable();
-  }
-
   /**
-   * Display succesful global message after clearing cart.
+   * Close clear cart modal dialog
    *
-   * @param success
+   * @param reason to close dialog
    */
-  addSuccessGlobalMessage(): void {
-    this.globalMessageService.add(
-      { key: 'clearCart.cartClearedSuccessfully' },
-      GlobalMessageType.MSG_TYPE_CONFIRMATION
-    );
-  }
-
-  onComplete(): void {
-    this.closeDialog('Close dialog after cart cleared successfully');
-    this.activeCartFacade
-      .getEntries()
-      .pipe(take(1))
-      .subscribe((entries) => {
-        if (entries.length === 0) {
-          this.addSuccessGlobalMessage();
-        }
-      });
-  }
-
   closeDialog(reason: string): void {
     this.launchDialogService.closeDialog(reason);
   }
+
+  /**
+   * Display global message after clearing cart.
+   * By default, only message displayed is of type `Success`. A negative scenario
+   * related to cart has been handled in the occ layer already.
+   * Success param recognizes result of action and allows custom error message as well.
```

Comment:
Review the method doc/comment because the method does not display an error  message.

Reply:
Hey Patrick are you refering to the section of the doc:  `Success param recognizes result of action and allows custom error message as well.`
If so, I mean that customers can override the behaviour to display an error message as well if success param != true